ush/python/pygfs/task/archive_vars.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env python3
+"""
+Archive Variables Task
+
+Overview
+--------
+This module provides variables needed by YAML templates for archiving verification
+(vrfy) data for GFS, GEFS, and GCAFS systems. File set generation logic (loops,
+conditionals, path construction) is handled by the YAML templates themselves.
+
+Architecture
+------------
+Python provides VARIABLES → YAML templates build FILE SETS
+
+Python Code Responsibilities:
+  - Compute cycle-specific variables (cycle_HH, cycle_YMDH, cycle_YMD, head)
+  - Calculate COM directory paths with grid loops (0p25, 0p50, 1p00)
+  - Extract configuration keys (RUN, DO_* flags, FHMAX*, etc.)
+  - Provide complete arch_dict to YAML templates
+
+YAML Template Responsibilities (parm/archive/*_arcdir.yaml.j2):
+  - Build file sets with source → destination mappings
+  - Handle loops (forecast hours, grids, basins)
+  - Apply conditionals (DO_* flags, MODE, RUN type)
+  - Create mkdir lists for directory creation
+
+Key Methods
+-----------
+get_all_yaml_vars():
+  Main entry point - collects all variables for YAML templates
+
+add_config_vars():
+  Extracts configuration keys and COM* variables (created in job scripts)
+
+_get_cycle_vars():
+  Computes cycle-specific variables (cycle_HH, cycle_YMDH, cycle_YMD, head)
+
+Logging
+-------
+All public operational methods are decorated with @logit(logger).
+"""
+import os
+from logging import getLogger
+from typing import Any, Dict
+from wxflow import Task, logit, to_YMD, to_YMDH
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+class ArchiveVrfy(Task):
```

Comment:
If this is not a task and just a class, then it should not inherit from a `task` class.